fvm/state/accounts.go

```diff
@@ -282,11 +273,113 @@ func (a *Accounts) SetCode(address flow.Address, code []byte) error {
 		return nil
 	}
 
-	a.ledger.Set(string(address.Bytes()), string(address.Bytes()), keyCode, code)
+	return a.setValue(address, true, keyCode, code)
+}
+
+// GetStorageUsed returns the amount of storage used in bytes by this account
+func (a *Accounts) GetStorageUsed(address flow.Address) (uint64, error) {
+	storageUsedRegister, err := a.getValue(address, false, storageUsedRegisterName)
+	if err != nil {
+		return 0, err
+	}
+
+	if len(storageUsedRegister) == 0 {
+		return 0, fmt.Errorf("account %s storage used is not initialized", address.Hex())
+	}
+
+	storageUsed := binary.LittleEndian.Uint64(storageUsedRegister)
+	return storageUsed, nil
+}
+
+func (a *Accounts) setStorageUsed(address flow.Address, used uint64) error {
+	buffer := make([]byte, 8)
+	binary.LittleEndian.PutUint64(buffer, used)
+	return a.setValue(address, false, storageUsedRegisterName, buffer)
+}
+
+// GetValue returns a value stored in address' storage
+func (a *Accounts) GetValue(address flow.Address, key string) (flow.RegisterValue, error) {
+	return a.getValue(address, false, key)
+}
+
+func (a *Accounts) getValue(address flow.Address, isController bool, key string) (flow.RegisterValue, error) {
+	if isController {
+		return a.ledger.Get(string(address.Bytes()), string(address.Bytes()), key)
+	} else {
```

Comment:
```suggestion
	}
```

You can just remove the `else` case here